pkg/sharding/ranges_test.go

```diff
@@ -388,3 +396,234 @@ func TestLogRanges_AllShards(t *testing.T) {
 		})
 	}
 }
+
+func Test_logRangesFromPath(t *testing.T) {
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name            string
+		args            args
+		want            Ranges
+		content         string
+		wantJSON        bool
+		wantYaml        bool
+		wantInvalidJSON bool
+		wantErr         bool
+	}{
+		{
+			name: "empty",
+			args: args{
+				path: "",
+			},
+			want:    Ranges{},
+			wantErr: true,
+		},
+		{
+			name: "empty file",
+			args: args{
+				path: "one",
+			},
+			want:    Ranges{},
+			wantErr: false,
+		},
+		{
+			name: "valid json",
+			args: args{
+				path: "one",
+			},
+			want: Ranges{
+				{
+					TreeID:     1,
+					TreeLength: 2,
+				},
+			},
+			wantJSON: true,
+			wantErr:  false,
+		},
+		{
+			name: "valid yaml",
+			args: args{
+				path: "one",
+			},
+			want: Ranges{
+				{
+					TreeID:     1,
+					TreeLength: 2,
+				},
+			},
+			wantYaml: true,
+			wantErr:  false,
+		},
+		{
+			name: "invalid json",
+			args: args{
+				path: "one",
+			},
+			want:            Ranges{},
+			wantInvalidJSON: true,
+			wantErr:         true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.args.path != "" {
+				f, err := os.CreateTemp("", tt.args.path)
+				if err != nil {
+					t.Fatalf("Failed to create temp file: %v", err)
+				}
+				defer os.Remove(f.Name())
+				switch {
+				case tt.wantJSON:
+					if err := json.NewEncoder(f).Encode(tt.want); err != nil {
+						t.Fatalf("Failed to encode json: %v", err)
+					}
+				case tt.wantYaml:
+					if err := yaml.NewEncoder(f).Encode(tt.want); err != nil {
+						t.Fatalf("Failed to encode yaml: %v", err)
+					}
+				case tt.wantInvalidJSON:
+					if _, err := f.WriteString("invalid json"); err != nil {
+						t.Fatalf("Failed to write invalid json: %v", err)
+					}
+				}
+				if _, err := f.Write([]byte(tt.content)); err != nil {
+					t.Fatalf("Failed to write to temp file: %v", err)
+				}
+				defer f.Close()
+				defer os.Remove(f.Name())
+				tt.args.path = f.Name()
+			}
+			got, err := logRangesFromPath(tt.args.path)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("logRangesFromPath() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("logRangesFromPath() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_updateRange(t *testing.T) {
+	type args struct {
+		ctx context.Context
+		r   LogRange
+	}
+	tests := []struct {
```

Comment:
what about testing a case where `wantErr` == false?

Reply:
I don't know what I could pass to this `func` not to return an error.
```
if err := root.UnmarshalBinary(resp.SignedLogRoot.LogRoot); err != nil {
			return LogRange{}, err
		}
```

I am mocking the call.